src/driver/mio/mod.rs

```diff
@@ -0,0 +1,251 @@
+#[doc(no_inline)]
+pub use std::os::fd::{AsRawFd, FromRawFd, IntoRawFd, RawFd};
+use std::{
+    cell::RefCell, collections::VecDeque, io, mem::MaybeUninit, ops::ControlFlow, time::Duration,
+};
+
+pub(crate) use libc::{sockaddr_storage, socklen_t};
+use mio::{
+    event::{Event, Source},
+    unix::SourceFd,
+    Events, Interest, Poll, Token,
+};
+use slab::Slab;
+
+use crate::driver::{Entry, Poller};
+
+pub(crate) mod op;
+
+/// Abstraction of operations.
+pub trait OpCode {
+    /// Perform the operation before submit, and return [`Decision`] to
+    /// indicate whether submitting the operation to mio is required.
+    fn pre_submit(&mut self) -> io::Result<Decision>;
+
+    /// Perform the operation after received corresponding
+    /// event.
+    fn on_event(&mut self, event: &Event) -> io::Result<ControlFlow<usize>>;
+}
+
+/// Result of [`OpCode::pre_submit`].
+pub enum Decision {
+    /// Instant operation, no need to submit
+    Completed(usize),
+    /// Async operation, needs to submit
+    Wait(WaitArg),
+}
+
+impl Decision {
+    /// Decide to wait for the given fd with the given interest.
+    pub fn wait_for(fd: RawFd, interest: Interest) -> Self {
+        Self::Wait(WaitArg { fd, interest })
+    }
+
+    /// Decide to wait for the given fd to be readable.
+    pub fn wait_readable(fd: RawFd) -> Self {
+        Self::wait_for(fd, Interest::READABLE)
+    }
+
+    /// Decide to wait for the given fd to be writable.
+    pub fn wait_writable(fd: RawFd) -> Self {
+        Self::wait_for(fd, Interest::WRITABLE)
+    }
+}
+
+/// Meta of mio operations.
+#[derive(Debug, Clone, Copy)]
+pub struct WaitArg {
+    fd: RawFd,
+    interest: Interest,
+}
+
+/// Low-level driver of mio.
+pub struct Driver(RefCell<DriverInner>);
```

Comment:
We've moved this RefCell to the runtime, and make the interface of `Poller` accepts `&mut self`. Remove the RefCell here please.